### PR TITLE
[YUNIKORN-25] Logging improvements

### PR DESCRIPTION
**Pull request by yangwwei**
This PR does

1) Remove unnecessary logs. Such as when predicate fails, we have pushed the info to the event system, we don't need to print the logging again and again. We would see some aggregated message like:

```
Warning  FailedScheduling  2m53s (x24 over 2m53s)  yunikorn                 [Predicate NodeUnknownCondition failed]
```
2) moving some of INFO level to DEBUG
3) Fix format of the event messages pushed to K8s.
4) Added a callback that handles the logging whenever the state changes for app/task

### CODE REVIEW

**Comment by yangwwei** [comment]
hi @wilfred-s 

Could you please help to review this PR?

File: pkg/cache/task.go [hunk 1]
```diff
@@ -282,8 +290,7 @@ func (task *Task) postTaskAllocated(event *fsm.Event) {
 			zap.String("podUID", string(task.pod.UID)))
 		if task.context.apiProvider.GetAPIs().VolumeBinder != nil {
 			if err := task.context.bindPodVolumes(task.pod); err != nil {
-				errorMessage = fmt.Sprintf("bind pod volumes failed, name: %s, uid: %s, %#v",
-					task.pod.Name, task.pod.UID, err)
+				errorMessage = fmt.Sprintf("bind pod volumes failed, name: %s, %#v", task.alias, err)
```
Comment (wilfred-s) [inline pkg/cache/task.go:293]: error does not implement GoStringer, why use `%#v`

File: pkg/cache/task.go [hunk 2]
```diff
@@ -296,8 +303,7 @@ func (task *Task) postTaskAllocated(event *fsm.Event) {
 			zap.String("podUID", string(task.pod.UID)))
 
 		if err := task.context.apiProvider.GetAPIs().KubeClient.Bind(task.pod, nodeID); err != nil {
-			errorMessage = fmt.Sprintf("bind pod failed, name: %s, uid: %s, %#v",
-				task.pod.Name, task.pod.UID, err)
+			errorMessage = fmt.Sprintf("bind pod volumes failed, name: %s, %#v", task.alias, err)
```
Comment (wilfred-s) [inline pkg/cache/task.go:306]: error does not implement GoStringer, why use `%#v`

File: pkg/cache/application.go [hunk 3]
```diff
@@ -103,6 +103,8 @@ func NewApplication(appID, queueName, user string, tags map[string]string, sched
 			string(events.RecoverApplication):  app.handleRecoverApplicationEvent,
 			string(events.RejectApplication):   app.handleRejectApplicationEvent,
 			string(events.CompleteApplication): app.handleCompleteApplicationEvent,
+			events.EnterState:                  app.enterState,
+
```
[the remark this answers is not in the document]
Comment (yangwwei) [reply pkg/cache/application.go:107]: Done

File: pkg/cache/task.go [hunk 4]
```diff
@@ -219,7 +222,7 @@ func (task *Task) handleSubmitTaskEvent(event *fsm.Event) {
 			log.Logger.Debug("updating pod state ",
 				zap.String("appID", task.applicationID),
 				zap.String("taskID", task.taskID),
-				zap.String("podName", fmt.Sprintf("%s/%s", task.pod.Namespace, task.pod.Name)),
+				zap.String("podName", fmt.Sprintf("%s", task.alias)),
```
[the remark this answers is not in the document]
Comment (yangwwei) [reply pkg/cache/task.go:225]: yeah, my bad.

[the remark this answers is not in the document]
Comment (yangwwei) [reply pkg/cache/task.go:225]: fixed

File: pkg/cache/task.go (hunk 1 above)
Comment (yangwwei) [reply pkg/cache/task.go:293]: fixed, we can use just string msg here.

File: pkg/cache/task.go (hunk 2 above)
Comment (yangwwei) [reply pkg/cache/task.go:306]: fixed

**Pull request closed without merging**